### ISSUE TITLE
Feature request: see N context lines around matches

### ISSUE THREAD

**Issue by kootenpv** [rcaloras/bashhub-client#30]
I recently missed a couple of times the -A/B/C argument of grep (but then for "bh"): finding lines around my target command (when I don't remember some new command but I do remember `ssh` was called before).

A: n lines before match
B: n lines after match
C: n lines before and after match

Example:
grep -r -C 5 "test" .

```
--
--
./new.py-
./new.py-from sqlalchemy import create_engine, MetaData, Table, Column
./new.py-from sqlalchemy import Integer, Unicode, UnicodeText
./new.py-from sqlalchemy import ForeignKey
./new.py-
./new.py:engine = create_engine("postgresql://postgres@localhost/test")
./new.py-
./new.py-metadata = MetaData()
./new.py-
./new.py-countries_table = Table("countries", metadata,
./new.py-                        Column("id", Integer, primary_key=True),
```

Notice the line with "test" and 5 lines before and after. It uses -- two times to delimit multiple matches. I'm indeed missing colors on the "test" match, I had to specifically add --color=always|auto flag. It would be nice to be the default to use colors I guess.

**Comment by kootenpv** [comment]
Wow, this has been a long time. I still find myself wanting this very much: just a little peek around a selected command. The better implementation would probably to use it as an extra option in interactive? After you've selected a command, enter a second selection of the commands around this.
It's very crucial, as sometimes you want to know for example to which directory you've moved or something before executing a command, or know what other steps you took.

Thanks my friend, still an active user of bashhub every day :)

**Comment by rcaloras** [comment]
Thanks for the encouragement! I've started work on the open-source backend. I'm sure this feature can be worked out :)

**Comment by kootenpv** [comment]
Still super relevant. A similar request to this is that I'd like to see all the commands on a single day. This really helps when you know 1 command that you'd like to use, or have used when installing something, but you just need to get all the context around it.

**Comment by DanEdens** [comment]
This doesn't need to be a bh feature, It is already available through grep.
Try this:
`bh | grep -c?`

For a single day, you can do something similar.

`➜  Dan.Edens bh -t | grep 2021-03
2021-03-17 08:19:29     bh test
2021-03-17 08:19:34     bh --help
2021-03-17 08:20:49     bh -t test
2021-03-17 08:22:57     bh
2021-03-17 08:23:26     grep -n
2021-03-17 08:23:29     grep --help
2021-03-17 08:24:55     bh -t
2021-03-17 08:25:09     bt | grep 2021-03
2021-03-17 08:25:16     bh | grep 2021-03`

**Comment by kootenpv** [comment]
> This doesn't need to be a bh feature, It is already available through grep.
> Try this:
> `bh | grep -c?`
> 
> For a single day, you can do something similar.
> '$ bh -t | grep 2021-03'

Only for the most recent though I think? How to do it for a command far in the past?

**Comment by DanEdens** [comment]
the timestamp is added by bashhub automatically when you enter it, so you would just put the date you want into grep like i did with 2021-03
'$ bh -t | grep 1995-03-15'

Just run 'bh -t' by itself and you should see what i mean

**Comment by rcaloras** [comment]
> Still super relevant. A similar request to this is that I'd like to see all the commands on a single day. This really helps when you know 1 command that you'd like to use, or have used when installing something, but you just need to get all the context around it.

@kootenpv Missed this comment. You can get similar functionality by searching by 'Session Id' which was released on 2.0.1. See: https://github.com/rcaloras/bashhub-client/releases/tag/2.0.1

```
-ses SESSION, --session SESSION
                        Filter by specific session id. Default is None.
                        Example: bh --session "d588e48b-f18e-4d47-b2ad-d24574b4d4d1"
```

Any chance this helps? Not the same as N lines of context, but can be used similarly trying to find the commands in a session when one was entered.

**Comment by kootenpv** [comment]
@DanEdens and @rcaloras thanks a lot, it's great to know next time I can easily then filter on timestamp/session!
